service/controller/app/v1/resource/chartoperator/resource.go

```diff
@@ -0,0 +1,153 @@
+package chartoperator
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/afero"
+	"gopkg.in/yaml.v2"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/helm/pkg/helm"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+)
+
+const (
+	Name = "chartoperatorv1"
+
+	defaultClusterDNSIP    = "172.31.0.10"
+	chartOperatorNamespace = "giantswarm"
+	chartOperatorRelease   = "chart-operator"
+	chartOperatorVersion   = "0.9.0"
+)
+
+// Config represents the configuration used to create a new clients resource.
+type Config struct {
+	// Dependencies.
+	Fs        afero.Fs
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+
+	// Settings.
+	RegistryDomain string
+}
+
+type Resource struct {
+	// Dependencies.
+	fs        afero.Fs
+	k8sClient kubernetes.Interface
+	logger    micrologger.Logger
+
+	// Settings.
+	registryDomain string
+}
+
+// New creates a new configured chartoperator resource.
+func New(config Config) (*Resource, error) {
+	if config.Fs == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Fs must not be empty", config)
+	}
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	if config.RegistryDomain == "" {
+		config.RegistryDomain = "quay.io"
+	}
+
+	r := &Resource{
+		// Dependencies.
+		fs:        config.Fs,
+		k8sClient: config.K8sClient,
+		logger:    config.Logger,
+
+		// Settings.
+		registryDomain: config.RegistryDomain,
+	}
+
+	return r, nil
+}
+
+func (r Resource) Name() string {
+	return Name
+}
+
+func (r Resource) installChartOperator(ctx context.Context, cr v1alpha1.App, helmClient helmclient.Interface) error {
+	var err error
+
+	var tarballPath string
+	{
+		tarballURL := fmt.Sprintf("https://giantswarm.github.io/giantswarm-catalog/chart-operator-%s.tgz", chartOperatorVersion)
```

Comment:
We should have a key function for this so we have the logic once and reuse it in the chart resource.